weblate/trans/models/translation.py

```diff
@@ -420,6 +421,16 @@ def check_sync(self, force=False, request=None, change=None) -> None:  # noqa: C
                                 unit.source = translated_unit.source
                         except UnitNotFoundError:
                             pass
+                    if (
+                        self.component.key_filter_re
+                        and re.match(self.component.key_filter_re, unit.context) is None
+                    ):
+                        self.log_info(
+                            "Doesn't match with key_filter, skipping: %s (%s)",
+                            unit,
```

Comment:
Include context here as that is what is being matched:

```suggestion
                            unit.context,
```